packages/segmented-control/src/SegmentedControlOption.tsx

```diff
@@ -338,9 +91,28 @@ export const SegmentedControlOption = forwardRef<
       didComponentMount.current = true;
     }, [focused, followFocus, usingKeyboard, isfocusInComponent]);
 
+    // Gets the number of children.
+    const childCount = React.Children.count(children);
+
+    // Wraps text in a `<span>`. This span will be used to clip text and add an ellipsis. We do this so that consumers don't have to add the `<span>` themselves. We can't add the ellipsis styles to the parent container because that container has `display: flex` which doesn't work with `text-overflow: ellipsis`.
+    // Also checks if child is an Icon.
+    const renderedChildren = useMemo(
+      () =>
+        React.Children.map(children, child => {
+          // if its a string wrap it in a span
+          if (typeof child === 'string')
+            return <span className={textEllipsisStyle}>{child}</span>;
+          // if its a LG icon or glyph set hasIcon to true
+          if (isComponentType(child, 'Icon') || isComponentGlyph(child))
+            setHasIcon(true);
```

Comment:
@TheSonOfThomp I had to make some tweaks to this. The icon color in the tab changes depending on if there is text or not. Since icons are added as children I added a check in the map and if there is an icon i set `setHasIcon` to `true` and use that to toggle icon only styles.